kong/plugins/http-log/schema.lua

```diff
@@ -42,6 +42,43 @@ return {
           }},
           { queue = typedefs.queue },
           { custom_fields_by_lua = typedefs.lua_code },
+          { enable_logged_headers_allow_list = {
+            type = "boolean",
+            default = false,
+            description = "When enabled, will ensure only headers from the logged_headers_allow_list are included in log payloads. All headers are logged if this is disabled.",
+          }},
+          { logged_headers_allow_list = {
+            type = "array",
+            elements = typedefs.header_name,
+            default = {
```

Comment:
I think this list doesn't cover all the standard HTTP headers, not even including the most commonly used set, `Accept-Charset`, `Connection`, and `Keep-Alive` for example.

The standard(registered) HTTP header list can be found at https://datatracker.ietf.org/doc/html/rfc4229.

There are some non-standard(registered) headers in this list(ipremoteaddr, ratelimit-*, x-original-*, x-request-start), these don't work on other users.

Regarding this list, I have a concern. CP(with this feature) will send all the headers to the DP(with this feature), if users provision a lot of`http-log` plugin instances, the configuration size may increase significantly, which may put too much pressure on the DP memory, since DP needs load all the configuration in the memory first(correct me if I'm wrong).

Another way is to define a separated list. For example

```lua
-- constants.lua under http-log plugin
local M = {
  registered_headers = { "Accept", "Accept-Encoding", ... } 
}

return M
```
-- schema.lua
{ logged_headers_allow_list = { type = "array", ..., default = { "registered_headers" } } }
-- ...
```
```lua

@hanshuebner Thought?

Reply:
@vm-001 Just to understand the concern:  The Control Plane will send a completely expanded configuration to the Data Plane, with all defaults filled in. This means that even if the feature is not used, all instances of the `http-log` plugin in the configuration will have the list of allowed headers attached.  If I'm understanding this correctly, then your concern is very valid.  I do not think, however, that a compiled-in list of standard headers is the right way to go.  Instead, I would propose that no default for `logged_headers_allow_list` is specified and that it would be an error to enable `enable_logged_headers_allow_list` without specifying `logged_headers_allow_list` as well.

Reply:
No default value for logged_headers_allow_list should address my concern.

Reply:
A default list for common headers was just a ease of use idea, I'm happy to apply the suggested changes of having no default list if that's the preferred approach.

Reply:
I've updated the default value to an empty array.